### PR TITLE
Add error checking for cooperative matrix use and scope parameters

### PR DESCRIPTION
**Pull request by MrSidims**
Use should be: MatrixA, MatrixB or Accumulator.
Scope must be at max Invocation (others are not supported
by the translator).

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
@vmaksimo  @VyacheslavLevytskyy please take a look

**Review by svenvh — APPROVED** [review]
LGTM for the KHR parts.

File: lib/SPIRV/libSPIRV/SPIRVType.cpp [hunk 1]
```diff
@@ -336,4 +336,22 @@ void SPIRVTypeCooperativeMatrixKHR::decode(std::istream &I) {
...
 }
 
+void SPIRVTypeCooperativeMatrixKHR::validate() const {
+  SPIRVEntry::validate();
+  SPIRVErrorLog &SPVErrLog = this->getModule()->getErrorLog();
+  SPIRVConstant *UseConst = static_cast<SPIRVConstant *>(this->getUse());
+  auto InstName = OpCodeNameMap::map(OC);
+  uint64_t UseValue = UseConst->getZExtIntValue();
+  SPVErrLog.checkError(
+      (UseValue <= internal::InternalJointMatrixUse::Accumulator),
```
Comment (vmaksimo) [inline lib/SPIRV/libSPIRV/SPIRVType.cpp:346]: Let's use the official enum for cooperative matrix https://github.com/KhronosGroup/SPIRV-Headers/blob/main/include/spirv/unified1/spirv.hpp#L1339

**Comment by VyacheslavLevytskyy** [comment]
looks good

**Pull request merged**